### ISSUE TITLE
Project does not build against Hive 0.12.0-cdh5.0.0-beta-2 and later

### ISSUE THREAD

**Issue by ivmaykov** [rcongiu/Hive-JSON-Serde#63]
Looks like this project does not build Hive versions based on 0.12.0. I tried both 0.12.0-cdh5.0.0-beta-2 and 0.12.0-cdh5.0.0. It does build against 0.11.0-cdh5.0.0-beta-1. Here is the output of the mvn package command:

```
$ mvn -Dcdh.version=0.12.0-cdh5.0.0-beta-2 package
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for org.openx.data:json-serde:jar:1.1.9.2
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 37, column 21
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building openx-json-serde 1.1.9.2
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ json-serde ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:compile (default-compile) @ json-serde ---
[INFO] Compiling 31 source files to /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringLongObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringTimestampObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringFloatObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringByteObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringDoubleObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/JsonStringJavaObjectInspector.java:[30,4] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringShortObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR]     constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
      (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
    constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
      (actual and formal argument lists differ in length)
/Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringIntObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[INFO] 8 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.491s
[INFO] Finished at: Wed Apr 16 15:00:52 PDT 2014
[INFO] Final Memory: 22M/310M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project json-serde: Compilation failure: Compilation failure:
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringLongObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringTimestampObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringFloatObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringByteObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringDoubleObjectInspector.java:[28,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/JsonStringJavaObjectInspector.java:[30,4] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringShortObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector(PrimitiveTypeInfo) is not applicable
[ERROR] (actual argument PrimitiveTypeEntry cannot be converted to PrimitiveTypeInfo by method invocation conversion)
[ERROR] constructor AbstractPrimitiveJavaObjectInspector.AbstractPrimitiveJavaObjectInspector() is not applicable
[ERROR] (actual and formal argument lists differ in length)
[ERROR] /Users/ilyam/src/github/rcongiu/Hive-JSON-Serde/src/main/java/org/openx/data/jsonserde/objectinspector/primitive/JavaStringIntObjectInspector.java:[29,8] error: no suitable constructor found for AbstractPrimitiveJavaObjectInspector(PrimitiveTypeEntry)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[Wed Apr 16 15:00:52] 1.9.2-p290 - 1.7.0_51 cdh5.0.0-beta-2 ilyam ~/src/github/rcongiu/Hive-JSON-Serde(master)
$
```

**Comment by dblock** [comment]
This should be closed via #64.

**Comment by rcongiu** [comment]
Fixed in #64
